lib/mps_statespace.h

```diff
@@ -155,6 +155,63 @@ class MPSStateSpace {
     }
   }
 
+  // Computes <state1 | state2 > for two equal sized MPS.
+  // Requires: state1.bond_dim() == state2.bond_dim() &&
+  //           state1.num_qubits() == state2.num_qubits()
+  static std::complex<fp_type> InnerProduct(MPS& state1, MPS& state2) {
```

Comment:
Perhaps add `const`: `MPS& state1` -> `const MPS& state1` and `MPS& state2` -> `const MPS& state2`?

Reply:
In the code I do write to the scratch spaces in both `state` variables so I can't make them const. Compiler complains about casting away constness if I do.